libraries/bts_7200_current_sense/src/bts_7200_current_sense.c

```diff
@@ -0,0 +1,60 @@
+#include "bts_7200_current_sense.h"
+
+static void prv_measure_current(SoftTimerId timer_id, void *context) {
+  Bts7200Storage *storage = context;
+
+  gpio_set_state(storage->select_pin, GPIO_STATE_LOW);
+  adc_read_raw(storage->sense_channel, &storage->reading_low);
+
+  gpio_set_state(storage->select_pin, GPIO_STATE_HIGH);
+  adc_read_raw(storage->sense_channel, &storage->reading_high);
+
+  storage->callback();
```

Comment:
your callback doesn't look like it's taking a *context, also pass in the readings in the callback's inputs as well.